modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/views/table/TableHeadCell.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayIcon from '@clayui/icon';
+import ClayTable from '@clayui/table';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+function TableHeadCell({
+	contentRenderer,
+	expand,
+	expandableColumns,
+	fieldName,
+	hideColumnLabel,
+	label,
+	sortable,
+	sorting,
+	sortingKey: sortingKeyProp,
+	updateSorting,
+}) {
+	const sortingKey =
+		sortingKeyProp || (Array.isArray(fieldName) ? fieldName[0] : fieldName);
+
+	const sortingMatch = sorting.find((element) => element.key === sortingKey);
+
+	function handleSortingCellClick(event) {
+		event.preventDefault();
+
+		const updatedSortedElements = sortingMatch
+			? sorting.map((element) =>
+					element.key === sortingKey
+						? {
+								...element,
+								direction:
+									element.direction === 'asc'
+										? 'desc'
+										: 'asc',
+						  }
+						: element
+			  )
+			: [
+					{
+						direction: 'asc',
+						fieldName,
+						key: sortingKey,
+					},
+			  ];
+
+		updateSorting(updatedSortedElements);
+	}
+
+	return (
+		<ClayTable.Cell
+			className={classNames(
+				contentRenderer && `content-renderer-${contentRenderer}`,
+				expandableColumns
+					? expand && 'table-cell-expand-small'
+					: 'table-cell-expand-smaller'
+			)}
+			headingCell
+			headingTitle
+		>
+			{sortable ? (
+				<a
+					className="inline-item text-nowrap text-truncate-inline"
+					data-senna-off
+					href="#"
+					onClick={handleSortingCellClick}
+				>
+					{(!hideColumnLabel && label) || ''}
+					<span className="inline-item inline-item-after sorting-icons-wrapper">
+						<ClayIcon
+							className={classNames(
+								'sorting-icon',
+								sortingMatch?.direction === 'asc' && 'active'
```

Comment:
Same as above, it's probably gonna be a bit more readable to use the object syntax for `classNames`. Note that you can use both an object and a string. So the first argument would be `'sorting-icon'` and the second arg would be an object for toggling `'active'`

Reply:
Only on this one and the other one below I would prefer keeping them as they are. Passing both a string and an object to `classnames` looks a bit less readable (or maybe I'm just not used to it). `active` also will not be highlighted as a string due to our linter an it looks a bit strange to me:

```
className={classNames(
	'sorting-icon',
	{
		active: sortingMatch?.direction === 'asc'
	}
)}
```

Btw if there are some standards we have to follow in this cases, not problems at all :)

Reply:
I don't think we have any standards written down, and we probably won't either unless we start to see `classNames()` usage "spiralling into chaos" 😂.

I don't have an opinion on this strong enough to make me want to comment, but seeing as the thread is already open, the way I usually tend to write these things to avoid the feeling of impedance mismatch you get when mixing string and object is to just use object for everything:

```jsx
<ClayIcon
    className={{
        active: sortingMatch?.direction === 'asc',
        'sorting-icon': true,
    }}
/>
```

but like all things, it's a trade-off; like you say @FabioDiegoMastrorilli, you lose the string highlighting of the `active` classname. We should be able to get it back though because I believe a long-standing Prettier bug has been fixed (see [discussion](https://github.com/liferay/eslint-config-liferay/pull/39)) that would format the above as:

```js
<ClayIcon
    className={{
        'active': sortingMatch?.direction === 'asc',
        'sorting-icon': true,
    }}
/>
```

Reply:
@FabioDiegoMastrorilli, thats fine, its not vital that you use the object notation, I just wanted to make sure you were aware of it.